modules/opengl/include/modules/opengl/image/imagecompositor.h

```diff
@@ -50,7 +52,13 @@ class IVW_MODULE_OPENGL_API ImageCompositor {
     void composite(ImageInport& source, ImageOutport& target, ImageType type);
 
 private:
+    void copyTextures(const Image& target);
+    void bindTextures(TextureUnitContainer& cont, const std::string& id);
+
     Shader shader_;
+    Texture2D colorTex_;
```

Comment:
is not the interface for `composite(...)` incorrect?
It is not obvious that it should combine source and target into target.

It should be `composite(const Image& first, const Image& second, Image& target, ImageType type);` in my opinion.
Not part of this PR, but the documentation about what the supplied fragment shader should take as input and produce is missing.

Reply:
This code was >5 years old and it broke when cleaning up the FBO state. That is two regression tests using the Geometry Picking processor (aka. `meshpicking.h`) were failing.
This is nothing more than a hot fix. I don't even know why we would need such a special compositing in the first place.
Even the shader looks like it does nothing special, so ActivateAndCopy might actually do the expected thing.

https://github.com/inviwo/inviwo/blob/master/modules/opengl/glsl/composite.frag

Reply:
where is it use can we remove it?